### DISCUSSION TITLE
How to return the same error message regardless of the specific error

### DISCUSSION THREAD

**Discussion by sacrosanctic** [open-circle/valibot#467] [General]
```ts
string("Please make a selection.", [
  minLength(1, "Please make a selection."),
])
```

Currently, I have to write out the same msg for every fn in the pipe.

**Comment by fabian-hiller** [comment]
You can overwrite the message globally with [`setGlobalMessage`](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6MA6VAUxgHFSIKyBZNqlTIA5mwAUhAAqk2aNnBDIA1ouRx2cgMYxgEAHbNCASgb1th1PADK2gBZtlcALyZmvAFZtdEgN70cHBOyMCkAFzu1lDABqISANosILEAMmxxMPYSAIwmADTuIWESJgC6BYFwYGioAO7QACaRLNGx8UnMKQbpmdl5hclpGaJZEgAc5ZUAvmaMlgbWcFBCAK6k8G6tyNhs0shQ7BJ2jsqFAUHFEUSeyAZsAAJsAB7I4HKE+VU1wg1QzUQcgAmADMABYAKwANgA7J96LNzAtUBA5Mw2FA8FAJCtUOsYGYgA) or [locally](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6CBQBWfJjAAUAb3pw4A5MFIAucRyjBWAc3kBtDGJAGAMnyMwe8gIwBKADTj1m+c4C6blXBgaKgA7tAAJjrmegbGZhbWtob2Tm7ilqw2dg4AHL7+AL7ODMxsHHBQfKgArqTwotHI2HwACshQqHzyAdz8gq4BimqCmjqEUsisfAACfAAeyOCkfITuQaihEWOOAEwAzAAsAKwAbADshHAFA6pDIFWoyIZ8Yy3LaHxwggDWn0KdyzkwDYYkuBXoxUYLHYEGWYj4UDwUHklRqdWKQA) when calling `parse` or `safeParse`.

**Reply by sacrosanctic** [reply]
can it be more local, like an error msg for email and one for password?

**Reply by fabian-hiller** [reply]
Ah, now I understand what you are looking for. If you always pass a string, you could remove the custom error message from `string` and just use one for the pipeline action. That's how I define my schemas.

**Comment by fabian-hiller** [comment]
Update:

Any schema and action has its own error message:

```ts
const RecordSchema = v.record(v.string('key error'), v.pipe(v.string('value error'), v.decimal('decimal error')), 'object error');
```

If you want the same error message for all nested schemas you can use the `config` method or write your own [`message` method](https://github.com/fabian-hiller/valibot/discussions/463#discussioncomment-9694410):

```ts
const RecordSchema = v.config(v.record(v.string(), v.string()), { message: 'error' });
```

If you want the same error message for all schemas when using `parse` or `safeParse` you can use the `config` argument:

```ts
const RecordSchema = v.record(v.string(), v.string());

const output = v.parse(RecordSchema, data, {  message: 'error'  });
```

If you want the same error message for all schemas globally you can use `setGlobalMessage`:

```ts
v.setGlobalMessage('error');
```

**Reply by sacrosanctic** [reply]
nice, i think `config` is what i'm looking for. I'll check it out.
